### ISSUE TITLE
[Semantics / Accessibility ] Expanded property doesn't announce anything 

### ISSUE THREAD

**Issue by SylberG** [flutter/flutter#138891]
### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/flutter/flutter/issues)
- [X] I have read the [guide to filing a bug](https://flutter.dev/docs/resources/bug-reports)

### Steps to reproduce

1. Run the provided code on a real device, simulator or emulator (iOS or Android).
2. Use TalkBack(Android), VoiceOver(iOS), Accessibility Inspector(iOS) on a real device, simulator or emulator.
3. Select the Text element with the following value: 'Expanded Property Test'.
4. Listen!

---
My setup:
Android Emulator: Pixel API 33 - Android 13.0 Google Play | arm64
iPhone Simulator: iPhone 14 Pro Max - iOS 17.0

### Expected results

I would expect that the 'Expanded' state would be announced, or something related to it.

### Actual results

Nothing is being announced, and can't see anything related to the Expanded state in the Accessibility Inspector.

### Code sample

<details><summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        colorScheme: ColorScheme.fromSeed(seedColor: Colors.deepPurple),
        useMaterial3: true,
      ),
      home: const MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  const MyHomePage({super.key, required this.title});

  final String title;

  @override
  State<MyHomePage> createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        backgroundColor: Theme.of(context).colorScheme.inversePrimary,
        title: Text(widget.title),
      ),
      body: Center(
        child: Semantics(
          expanded: true,
          child: const Text('Expanded Property Test'),
        ),
      ),
    );
  }
}

```

</details>

### Screenshots or Video

<details>
<summary>Screenshots / Video demonstration</summary>

[Expanded Property Test Video.webm](https://github.com/flutter/flutter/assets/50016143/47dc1aa3-1041-4be6-9942-67a2fccd0b9c)

![Screenshot 2023-11-22 at 13 17 48](https://github.com/flutter/flutter/assets/50016143/86e1b23e-ae79-4ead-9659-ec722d2eaf96)

</details>

### Flutter Doctor output

<details><summary>Doctor output</summary>

```console
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.16.0, on macOS 14.1.1 23B81 darwin-arm64, locale en-NL)
[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.2)
[✓] Xcode - develop for iOS and macOS (Xcode 15.0.1)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2022.3)
[✓] VS Code (version 1.84.2)
[✓] Connected device (3 available)
[✓] Network resources

• No issues found!
```

</details>

**Comment by dam-ease** [comment]
Thanks for filing this. 
I see this is a known issue being tracked here https://github.com/flutter/flutter/issues/92040.
Kindly upvote, comment and follow up on that issue for updates.
Closing this in favour of the already existing issue, if you disagree, please comment and I'll reopen this.
Thank you!

**Comment by SylberG** [comment]
> Thanks for filing this. I see this is a known issue being tracked here #92040. Kindly upvote, comment and follow up on that issue for updates. Closing this in favour of the already existing issue, if you disagree, please comment and I'll reopen this. Thank you!

Hi @dam-ease, I'll disagree because the proposal has been implemented, the expanded parameter has been added to the Semantics widget. But in the end it doesn't do nothing, and it doesn't announce nothing..

**Comment by iamwalter** [comment]
I have running into the same issue. It would be nice this would be worked on.

**Comment by dam-ease** [comment]
Thanks for your responses.
From my understanding of this, I think there are two parts to this previously open proposal

> I would like the Semantics widget to have an expanded parameter so that it announces the state for VoiceOver and TalkBack.

I see that although the widget seems to have an expanded parameter, the state announcement hasn't been implemented, my guess of which might be the reason this is still open, hence why I marked this a duplicate.

Could you help with more information on other differences in this proposal I might have missed out on?

**Comment by SylberG** [comment]
Thank you for re-opening this issue.

I see that only 1 part of the issue is implemented, which is adding the ‘expanded’ argument to the Semantics widget. However, it seems to me that there is no implementation for this argument because no element metadata is changing.

**Comment by dam-ease** [comment]
Thanks for your response.

> However, it seems to me that there is no implementation for this argument because no element metadata is changing.

Please can you help with more information on this?

**Comment by SylberG** [comment]
I'm expressing that when I set the parameter to either true or false, there is no observable change. This leaves me unsure whether this lack of effect is due to an issue or if the feature is not implemented.

**Comment by dam-ease** [comment]
Thanks for your response. 
I can reproduce this on my end as well.
This could still be related/similar to https://github.com/flutter/flutter/issues/92040.
I'll be labelling for more insights from the team.
Thank you!
<details><summary>flutter doctor -v</summary>
<p>

```
[!] Flutter (Channel stable, 3.16.0, on macOS 14.0 23A344 darwin-arm64, locale
    en-NG)
    • Flutter version 3.16.0 on channel stable at
      /Users/damilolaalimi/sdks/flutter
    ! Warning: `dart` on your path resolves to
      /opt/homebrew/Cellar/dart/3.1.5/libexec/bin/dart, which is not inside your
      current Flutter SDK checkout at /Users/damilolaalimi/sdks/flutter.
      Consider adding /Users/damilolaalimi/sdks/flutter/bin to the front of your
      path.
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision db7ef5bf9f (9 days ago), 2023-11-15 11:25:44 -0800
    • Engine revision 74d16627b9
    • Dart version 3.2.0
    • DevTools version 2.28.2
    • If those were intentional, you can disregard the above warnings; however
      it is recommended to use "git" directly to perform update checks and
      upgrades.

[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    • Android SDK at /Users/damilolaalimi/Library/Android/sdk
    • Platform android-34, build-tools 34.0.0
    • ANDROID_HOME = /Users/damilolaalimi/Library/Android/sdk
    • Java binary at: /Applications/Android
      Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      17.0.6+0-17.0.6b802.4-9586694)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 15.0.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 15A507
    • CocoaPods version 1.12.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2022.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build
      17.0.6+0-17.0.6b802.4-9586694)

[!] Android Studio (version unknown)
    • Android Studio at /Users/damilolaalimi/Downloads/Android Studio
      Preview.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    ✗ Unable to determine Android Studio version.
    • Java version OpenJDK Runtime Environment (build
      17.0.7+0-17.0.7b1000.6-10550314)

[✓] VS Code (version 1.84.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.50.0

[✓] VS Code (version 1.83.1)
    • VS Code at /Users/damilolaalimi/Downloads/Visual Studio Code.app/Contents
    • Flutter extension version 3.50.0

[✓] Connected device (4 available)
    • TECNO CC7 (mobile)         • 05199339BL004045          • android-arm64  •
      Android 9 (API 28)
    • Damilola’s iPhone (mobile) • 00008110-001964480AE1801E • ios            •
      iOS 17.1.1 21B91
    • macOS (desktop)            • macos                     • darwin-arm64   •
      macOS 14.0 23A344 darwin-arm64
    • Chrome (web)               • chrome                    • web-javascript •
      Google Chrome 119.0.6045.159

[✓] Network resources
    • All expected network resources are available.

! Doctor found issues in 2 categories.
``` 

</p>
</details>

**Comment by goderbauer** [comment]
cc @chunhtai @hangyujin

**Comment by lore-co** [comment]
Issue is still present

**Comment by mattyzenny** [comment]
To provide additional evidence, this issue is still reproducible on iOS with Flutter 3.44.4.

A native SwiftUI `DisclosureGroup` exposes an **Expanded** attribute in Accessibility Inspector, and VoiceOver announces its expanded/collapsed state. For the equivalent Flutter control, `Semantics(expanded: ...)` does not expose the **Expanded** attribute in Accessibility Inspector.

This suggests that Flutter’s expanded semantics state is not being mapped to the corresponding native iOS accessibility representation.



<details>
<summary>flutter doctor -v</summary>

```text
Flutter (Channel stable, 3.44.4, on macOS 26.5.1 25F80 darwin-arm64,
    locale en-US)
    • Flutter version 3.44.4 on channel stable at /opt/homebrew/share/flutter
    • Framework revision ad70ec4617
    • Engine revision a10d8ac38d
    • Dart version 3.12.2
    • DevTools version 2.57.0

[✓] Xcode - develop for iOS and macOS (Xcode 26.6)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 17F113
    • CocoaPods version 1.16.2

[✓] Connected device
    • iPhone 17 Pro (mobile) • [REDACTED] • ios
      • com.apple.CoreSimulator.SimRuntime.iOS-26-5 (simulator)
```

</details>
